File: leo2-common/src/main/java/org/deku/leo2/rest/entities/v1/RoutingRequest.java
package org.deku.leo2.rest.entities.v1;

import com.wordnik.swagger.annotations.ApiModel;
import com.wordnik.swagger.annotations.ApiModelProperty;
import org.deku.leo2.rest.entities.ShortDate;

/**
 * Created by masc on 23.06.15.
 */
@ApiModel(description = "Routing request")
public class RoutingRequest {
    private Float mWeight;
    private Integer mServices;
    private requestParticipant mSender;
    private requestParticipant mConsignee;
    private ShortDate mSendDate;
    private ShortDate mDeliveryDate;

    public RoutingRequest() { }


    @ApiModelProperty(dataType = "date", example = "2015-06-01", position = 10, required = true, value = "Senddate", allowableValues = "2015-06-01")
    public ShortDate getSendDate() {
        return mSendDate;
    }

    public void setSendDate(ShortDate sendDate) {
        mSendDate = sendDate;
    }

    @ApiModelProperty(dataType = "date", example = "2015-06-02", position = 20, required = false, value = "Deliverydate", allowableValues = "2015-06-02")
    public ShortDate getDeliveryDate() {
        return mDeliveryDate;
    }

    public void setDeliveryDate(ShortDate deliveryDate) {
        mDeliveryDate = deliveryDate;
    }


    @ApiModelProperty(value="Sum of DeKu Servicvalues", position = 22, required = false)
    public void setServices(Integer services) {
        mServices = services;
    }

    public Integer getServices() {
        return mServices;
    }

    @ApiModelProperty(value="Real weight", position = 24, required = false)
    public Float getWeight() {
        return mWeight;
    }

    public void setWeight(Float weight) {
        mWeight = weight;
    }


    @ApiModelProperty(value="Sender", position = 30, required = false)
    public requestParticipant getSender() {
        return mSender;
    }

    public void setSender(requestParticipant sender) {
        mSender = sender;
    }

    @ApiModelProperty(value="Consignee", position = 40, required = false)
    public requestParticipant getConsignee() {
        return mConsignee;
    }

    public void setConsignee(requestParticipant consignee) {
        mConsignee = consignee;
    }


}

File: leoz-common/src/main/java/org/deku/leoz/rest/entities/v1/RoutingRequest.kt
package org.deku.leoz.rest.entities.v1

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import org.deku.leoz.rest.entities.ShortDate

/**
 * Created by masc on 23.06.15.
 */
@ApiModel(value = "RoutingRequest", subTypes = arrayOf(RoutingRequest.RequestParticipant::class), description = "Routing request")
class RoutingRequest {
    /**
     * Sender or consignee attributes
     * Created by masc on 23.06.15.
     */
    @ApiModel(value = "RequestParticipant", description = "Request participant. Sender or consignee")
    class RequestParticipant {
        @ApiModelProperty(dataType = "string", example = "DE", value = "Country two-letter ISO-3166", position = 10, required = true)
        var country: String? = null

        @ApiModelProperty(dataType = "string", example = "36286", value = "Zip code accordant to country specification", position = 20, required = true)
        var zip: String? = null

        @ApiModelProperty(dataType = "string", example = "09:00", position = 40, required = false, value = "Time window (from)", allowableValues = "00:00 - 23:59")
        var timeFrom: String? = null

        @ApiModelProperty(dataType = "string", example = "12:00", position = 50, required = false, value = "Time window (to)", allowableValues = "00:00 - 23:59")
        var timeTo: String? = null

        @ApiModelProperty(dataType = "string", example = "020", position = 60, required = false, value = "Desired station number", allowableValues = "010 - 999")
        var desiredStation: String? = null

        constructor() {  }
    }

    @ApiModelProperty(dataType = "date", example = "2015-06-01", position = 10, required = true, value = "Send date", allowableValues = "2015-06-01")
    var sendDate: ShortDate? = null

    @ApiModelProperty(dataType = "date", example = "2015-06-02", position = 20, required = false, value = "Desired delivery date", allowableValues = "2015-06-02")
    var desiredDeliveryDate: ShortDate? = null

    @ApiModelProperty(value = "Sum of DeKu servic values", position = 22, required = false)
    var services: Int? = null

    @ApiModelProperty(value = "Real weight", position = 24, required = false)
    var weight: Float? = null

    @ApiModelProperty(value = "Sender", position = 30, required = false)
    var sender: RequestParticipant? = null

    @ApiModelProperty(value = "Consignee", position = 40, required = false)
    var consignee: RequestParticipant? = null
}
